### ISSUE TITLE
Need to make sure authenticated users are properly redirected to local translation

### ISSUE THREAD

**Issue by mikefantini** [DoSomethingArchive/legacy-website#5611]
If 
- US EN translation is published
- GL EN translation exists, but is unpublished
- When US user goes to thor.dosomething.org/campaigns/gtfo, user sees US EN content, but URL remains thor.dosomething.org/campaigns/gtfo (not: /us/campaigns/gtfo). I don't _think_ this matters, but not sure. When I click on Explore Campaigns from this view I get taken to /us/campaigns.

If 
- US EN translation is published
- GL EN translation doesn't exist
- When US user goes to thor.dosomething.org/campaigns/gtfo, users sees US EN content, but URL remains thor.dosomething.org/campaigns/gtfo (not: /us/campaigns/gtfo). Again, I don't _think_ this matters, but not sure.

If
- US EN translation is published
- GL EN translation is published
- When US user goes to thor.dosomething.org/campaigns/gtfo, user see GL EN content.

@mshmsh5000 mentioned some of this was configured here https://github.com/DoSomething/phoenix/blob/global/lib/modules/dosomething/dosomething_global/dosomething_global.module#L72

I'm not sure if it matters that the user stays on thor.dosomething.org/campaigns/gtfo instead of going to thor.dosomething.org/us/campaigns/gtfo. If it does, then maybe this should be a BLOCKER. If not, then it probably doesn't need to be.

**Comment by angaither** [comment]
not sure I understand this issue... it sounds like we want to change the logic for US users now?

**Comment by mshmsh5000** [comment]
I'm going to rephrase this as a use case that we need to satisfy.

If
- I am an authenticated user in a supported country (currently US, MX, BR), and
- I access the global URL of a campaign that has a published version in my country, then
- I will be explicitly redirected to my country's version's URL.

Example:
- I am an authenticated MX user.
- I access `/campaigns/gtfo`.
- There is a published MX version of this campaign at `/mx/campañas/gtfo`.
- I get a `302 Found` redirect to `/mx/campañas/gtfo`.

This is the same behavior the system currently exhibits for anonymous requests. Previously, we'd said that we would never redirect _authenticated_ requests in this way. But we've realized that we actually need to, in order to offer a coherent experience to users who are logged in to the system when they click on a link in, say, a Facebook post.

**Comment by angaither** [comment]
ooooooooohkay, got it, thanks @mshmsh5000!

**Comment by mikefantini** [comment]
Sorry wanted to get the issue in quickly before I had to run to a meeting, but yes that is the desired outcomes.

**Comment by itsjoekent** [comment]
this bug isn't occurring because of a logic problem in the redirect stuffs (yay), it's happening because this line of logic is incorrectly blocking the redirect from happening https://github.com/DoSomething/phoenix/blob/global/lib/modules/dosomething/dosomething_global/dosomething_global.module#L64
